site/static/examples/depends-on/depends-on.yaml

```diff
@@ -0,0 +1,61 @@
+# This example creates three ReplicatedJobs.
+# The launcher ReplicatedJob waits until the initializer ReplicatedJob is Complete.
+# After that, the worker ReplicatedJob waits until the launcher ReplicatedJob is Ready.
+# Launcher ReplicatedJob will be in Ready status once InitContainer is finished.
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: depends-on
+spec:
+  replicatedJobs:
+    - name: initializer
+      template:
+        spec:
+          template:
+            spec:
+              containers:
+                - name: initializer
+                  image: busybox
+                  command:
+                    - /bin/sh
+                    - -c
+                    - "echo 'initializer runs for 10 seconds' && sleep 10"
+    - name: launcher
```

Comment:
For the launcher maybe we should have some kind of readiness probe that can explain how this would be used for ready?

Right now it would go to active/running very quick due to no readiness probe.

Reply:
Sure, I am changed to startupProbe for launcher.

>Right now it would go to active/running very quick due to no readiness probe.

Actually, the Job will be Ready only after all InitContainers are complete: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#:~:text=Ready%3A%20the%20Pod%20is%20able%20to%20serve%20requests%20and%20should%20be%20added%20to%20the%20load%20balancing%20pools%20of%20all%20matching%20Services.

Reply:
> Ready: the Pod is able to serve requests and should be added to the load balancing pools of all matching Services.

All init containers finished would have the condition Initialized.